sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -405,6 +405,11 @@ object CatalogTypes {
    * Specifications of a table partition. Mapping column name to column value.
    */
   type TablePartitionSpec = Map[String, String]
+
+  /**
+   * Initialize an empty spec.
+   */
+  lazy val emptyTablePartitionSpec: TablePartitionSpec = Map.empty[String, String]
```

Comment:
`Map.empty` is already an object, I think we can jus inline it

Reply:
We wanted to refer the val `emptyTablePartitionSpec` as `TablePartitionSpec`, not `Map[String, String]`, though they are equal.